example_dll build failing

I went through all of the steps described in the How To Build wiki page.

Once I fixed the _WIN32 header issue for the sleep command I was greeted with 47 errors that look like this:
```
F:\VisualStudio2015\VC\include\memory(901): error C2664: 'intercept::client::types::text_ptr::text_ptr(const intercept::client::types::text_ptr &)': cannot convert argument 1 from 'intercept::rv_types::game_data *' to 'intercept::rv_types::game_value'
```
They are all coming from the example_dll.
Any suggestions?

--- a/src/client/intercept/client/sqf/common_helpers.cpp
+++ b/src/client/intercept/client/sqf/common_helpers.cpp
@@ -27,47 +27,47 @@ namespace intercept {
 
 			object __retrieve_nular_object(nular_function fnc_) {
 				game_value function_return_number = host::functions.invoke_raw_nular(fnc_);
-				return std::make_shared<object_ptr>(function_return_number.data);
+				return std::make_shared<object_ptr>(function_return_number);
 			}
 
 			text __retrieve_nular_text(nular_function fnc_) {
 				game_value function_return_number = host::functions.invoke_raw_nular(fnc_);
-				return std::make_shared<text_ptr>(function_return_number.data);
+				return std::make_shared<text_ptr>(function_return_number);
 			}
 
 			rv_namespace __retrieve_nular_namespace(nular_function fnc_) {
 				game_value function_return_number = host::functions.invoke_raw_nular(fnc_);
-				return std::make_shared<rv_namespace_ptr>(function_return_number.data);
+				return std::make_shared<rv_namespace_ptr>(function_return_number);
 			}
 
 			location __retrieve_nular_location(nular_function fnc_) {
 				game_value function_return_number = host::functions.invoke_raw_nular(fnc_);
-				return std::make_shared<location_ptr>(function_return_number.data);
+				return std::make_shared<location_ptr>(function_return_number);
 			}
 
 			side __retrieve_nular_side(nular_function fnc_) {
 				game_value function_return_number = host::functions.invoke_raw_nular(fnc_);
-				return std::make_shared<side_ptr>(function_return_number.data);
+				return std::make_shared<side_ptr>(function_return_number);
 			}
 
 			control __retrieve_nular_control(nular_function fnc_) {
 				game_value function_return_number = host::functions.invoke_raw_nular(fnc_);
-				return std::make_shared<control_ptr>(function_return_number.data);
+				return std::make_shared<control_ptr>(function_return_number);
 			}
 
 			group __retrieve_nular_group(nular_function fnc_) {
 				game_value function_return_number = host::functions.invoke_raw_nular(fnc_);
-				return std::make_shared<group_ptr>(function_return_number.data);
+				return std::make_shared<group_ptr>(function_return_number);
 			}
 
 			config __retrieve_nular_config(nular_function fnc_) {
 				game_value function_return_number = host::functions.invoke_raw_nular(fnc_);
-				return std::make_shared<config_ptr>(function_return_number.data);
+				return std::make_shared<config_ptr>(function_return_number);
 			}
 
 			display __retrieve_nular_display(nular_function fnc_) {
 				game_value function_return_number = host::functions.invoke_raw_nular(fnc_);
-				return std::make_shared<display_ptr>(function_return_number.data);
+				return std::make_shared<display_ptr>(function_return_number);
 			}
 
 			vector3 __retrieve_nular_vector3(nular_function fnc_) {
@@ -184,6 +184,11 @@ namespace intercept {
 				host::functions.invoke_raw_unary(fnc_, game_value_bool(val_));
 			}
 
+			void __empty_unary_control(unary_function fnc_, control ctrl_)
+			{
+				host::functions.invoke_raw_unary(fnc_, *ctrl_);
+			}
+
 			bool __bool_unary_object(unary_function fnc_, object obj_)
 			{
 				game_value bool_ret = host::functions.invoke_raw_unary(fnc_, *obj_);
@@ -275,7 +280,7 @@ namespace intercept {
 			object __object_unary_object(unary_function fnc_, object obj_)
 			{
 				game_value obj_ret = host::functions.invoke_raw_unary(fnc_, *obj_);
-				return std::make_shared<object_ptr>(obj_ret.data);
+				return std::make_shared<object_ptr>(obj_ret);
 			}
 		}
 	}

--- a/src/client/intercept/client/sqf/misc.cpp
+++ b/src/client/intercept/client/sqf/misc.cpp
@@ -1168,10 +1168,6 @@ namespace intercept {
             return __helpers::__bool_unary_string(client::__sqf::unary__deletestatus__string__ret__bool, value_);
         }
 
-        void delete_vehicle(object value_) {
-            __helpers::__empty_unary_object(client::__sqf::unary__deletevehicle__object__ret__nothing, value_);
-        }
-
         void detach(object value_) {
             __helpers::__empty_unary_object(client::__sqf::unary__detach__object__ret__nothing, value_);
         }
@@ -2142,10 +2138,6 @@ namespace intercept {
 
         float rad(float value_) {
             return __helpers::__number_unary_number(client::__sqf::unary__rad__scalar_nan__ret__scalar_nan, value_);
-        }
-
-        float random(float value_) {
-            return __helpers::__number_unary_number(client::__sqf::unary__random__scalar_nan__ret__scalar_nan, value_);
         }
 
         std::string rank(object value_) {

--- a/src/client/intercept/client/sqf/misc.hpp
+++ b/src/client/intercept/client/sqf/misc.hpp
@@ -324,7 +324,6 @@ namespace intercept {
         void delete_marker_local(std::string value_);
         void delete_site(object value_);
         bool delete_status(std::string value_);
-        void delete_vehicle(object value_);
         void detach(object value_);
         bool did_jipowner(object value_);
         bool difficulty_enabled(std::string value_);

